NeuroID/src/main/java/com/neuroid/tracker/NeuroID.kt

```diff
@@ -263,7 +306,16 @@ class NeuroID private constructor(
         NIDServiceTracker.rndmId = ""
     }
 
-   open fun start() {
+    open fun start() {
+        if (clientKey == "") {
+            NIDLog.e(
+                "NeuroID Error",
+                "Missing Client Key - please call configure prior to calling start"
+            )
+            throw IllegalArgumentException("NeuroID SDK Missing Client API Key");
```

Comment:
Is there a more appropriate exception?  There are no arguments to `start`.

Reply:
maybe IllegalStateException might be better.